### ISSUE TITLE
Neovim get stuck when I hit any Fn key.

### ISSUE THREAD

**Issue by Boltzmachine** [neovim/neovim#12386]
when I am in NORMAL mode, I hit the Fn keys(F1-F10), every things goes well. But when I am in INSERT mode, Neovim get stuck or suspended when I hit any Fn key. I tried many methods(such as C-q) but in vain.

Operating system/version: macOS/10.15.3
Terminal name/version: ITerm2. Happened both in tmux or not

`nvim --version`
```
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-param
eter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_
DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/config -I/tmp/neovim-20191107-85913-1wbgqz
6/neovim-0.4.3/src -I/usr/local/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/deps-build/include -I/usr/local/opt/gettext/include -I/Applications/Xcode.app/Contents/D
eveloper/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/src/nvim/auto -I/tmp/neovim-20191107-85913-1
wbgqz6/neovim-0.4.3/build/include
Compiled by brew@Catalina.local

Features: +acl +iconv +tui
See ":help feature-compile"
```

**Comment by teto** [comment]
but the rest of the system works (since catalina shows many system freeze) ? could you try with master ?

**Comment by clason** [comment]
I can reproduce this (fresh build from current master, macOS 15.5.5) -- not with F1, which triggers help, but with F5.

**Comment by weikang-q** [comment]
> I can reproduce this (fresh build from current master, macOS 15.5.5, and also on openSUSE Tumbleweed) -- not with F1, which triggers help, but with F5. This is through iTerm2 and builtin Terminal.
> 
> vim correctly inserts `<F5>`, as does **`nvim -U NONE`**. In fact, Terminal shows the hanging process to be `Python`, so I assume the problem is with `pynvim` (or the plugin itself, most likely Ultisnips; I didn't bisect my init.vim).

That's it! Neovim can correctly insert the Fns after I disable the Utilsnips plugin. But how to fix it? Is it an issue/bug of Utilsnips, pynvim or others?

**Comment by glacambre** [comment]
On 2020-05-28 03:00, qiuweikang wrote:
> That's it! Neovim can correctly insert the Fns after I disable the 
> Utilsnips plugin. But how to fix it? Is it an issue/bug of Utilsnips, 
> pynvim or others?

Try upgrading pynvim, this was sufficient to make the problem go away 
for https://github.com/SirVer/ultisnips/issues/12272, which sounds very 
similar to your problem.

This issue is probably a duplicate of 
https://github.com/neovim/neovim/issues/12293 .

**Comment by weikang-q** [comment]
I've already upgraded pynvim to the latest version(0.4.1). But the problem still exists.

**Comment by clason** [comment]
> Try upgrading pynvim, this was sufficient to make the problem go away
for https://github.com/SirVer/ultisnips/issues/12272, which sounds very
similar to your problem.

That fix isn't in a release yet. Might be good to make a bugfix release soon? @jamessan @Shougo

**Comment by erw7** [comment]
This problem should be fixed in pynvim 0.4.2. Therefore, this is closed.
